widget/popup_test.go

```diff
@@ -273,6 +276,56 @@ func TestPopUp_Stacked(t *testing.T) {
 	assert.Empty(t, test.Canvas().Overlays().List())
 }
 
+func TestPopUp_Layout(t *testing.T) {
+	win := test.NewWindow(NewLabel("OK"))
+	defer win.Close()
+	win.Resize(fyne.NewSize(80, 80))
+	pos := fyne.NewPos(6, 9)
+	size := fyne.NewSize(50, 40)
+
+	for name, tt := range map[string]struct {
+		pad        bool
+		castShadow bool
+	}{
+		"with padding with shadow":       {true, true},
+		"without padding with shadow":    {false, true},
+		"with padding without shadow":    {true, false},
+		"without padding without shadow": {false, false},
+	} {
+		t.Run(name, func(t *testing.T) {
+			content := NewLabel("Hi")
+			pop := newPopUp(content, win.Canvas())
+			pop.NotPadded = !tt.pad
+			pop.hideShadow = !tt.castShadow
+			pop.ShowAtPosition(pos)
+			defer test.Canvas().Overlays().Remove(pop)
+
+			pop.Resize(size)
+			r := cache.Renderer(pop)
+			expectedObjectCount := 2
+			if tt.castShadow {
+				expectedObjectCount++
+			}
+			require.GreaterOrEqual(t, len(r.Objects()), expectedObjectCount)
+
+			bgIdx := 0
+			if tt.castShadow {
```

Comment:
This doesn't test the no-shadow state does it?

Reply:
not explicitly, good catch
if the pop-up would cast a shadow with `hideShadow == true`, the following assertion regarding the background would fail. I’ll add an explicity assertion to clarify the intention.